check_ssl.sh

```diff
@@ -92,19 +92,33 @@ DATE_ACTUALLY_SECONDS=$(date +"%s")
 # GET CERTIFICATE |
 #------------------
 if [[ -z "${PROTOCOL}" ]]; then
-	HOST_CHECK=$(openssl s_client -servername "${HOST}" -connect "${IP}":"${PORT}" 2>&- | openssl x509 -enddate -noout)
+	HOST_CHECK=$(openssl s_client -servername "${HOST}" -connect "${IP}":"${PORT}" 2>&- | openssl x509 -enddate -subject -noout)
```

Comment:
Maybe we should use two calls to `openssl x509`. One for `openssl x509 -enddate -noout` for the date and another one for `openssl x509 -subject` would make handling of regular expressions easier.

I would recommend to start with the following structure:
```
CERTIFICATE=$(openssl s_client -servername "${HOST}" -connect "${IP}":"${PORT}" 2>&- | openssl x509)
END_DATE=$(echo "${CERTIFICATE}" | openssl x509 -noout -enddate)
SUBJECT=$(echo "${CERTIFICATE}" | openssl x509 -noout -subject)

Reply:
This would certainly give us cleaner looking code but I think this suggestion is not beneficial from a performance standpoint.

Writing the complete public key to a variable and calling the openssl binary on it twice is relatively costly compared to running a few grep's on short text strings in a pipe.

Reply:
My idea was to make the grep easier to understand. I'll have another look to make sure that we're not introducing any new edge cases.